Assets/Scripts/RicochetBullet.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RicochetBullet : MonoBehaviour
+{
+    private float damage = 1;
+    private int reflectionsRemaining = 5;
+    public Rigidbody2D rb;
+
+    void Start()
+    {
+        rb = this.gameObject.GetComponent<Rigidbody2D>();
+    }
+
+    private void OnCollisionEnter2D(Collision2D other)
+    {
+        if (other.gameObject.layer == layer.Entity)
```

Comment:
I'm not exactly sure if we really are using layers. So instead of checking layers what you can do is `Entity entity = other.gameObject.GetComponent<Entity>();` and then check if entity is null, from there we know that it isn't an entity, and the proceed to do the reflect code